page/learningchoice.jsx

```diff
@@ -24,18 +28,66 @@ css(`
 }
 `)
 
-export const LearningChoice = () => (
-  <Div class="div-learning">
-    <P class="title-learning">What do you want to learn 📚 ?</P>
-    {Object.entries(roles).map(([key, { name, color }]) => (
-      <span class="navlink-learning" key={key}>
-        <NavLink
-          href={`${API}/link/discord?speciality=${key}`}
-          style={{ color: parseColor(color), fontSize: 35 }}
+export const LearningChoice = () => {
+  let [errors, setErrors] = useState({})
+  if (!user) {
+    navigate('/login')
+    return
+  }
+  const send = (e) => {
+    e.preventDefault()
+    const form = new FormData(e.target)
+    const { why, speciality } = Object.fromEntries(form)
+    if (!why) {
+      setErrors((d) => {
+        return { ...d, why: 'you must fill this input' }
+      })
+      return
+    }
+    location.href = `${API}/link/discord?speciality=${speciality}`
+  }
+
+  return (
+    <Div class="div-learning">
+      <Form
+        submit="Go"
+        onSubmit={send}
+        style={{ textAlign: 'center' }}
+        title="Formulaire"
+      >
+        <Input
+          name="why"
+          comment="why do you want to join NaN?"
+          errors={errors}
+          type="text"
+        />
+        <br />
+        <br />
+        <Select
+          name="speciality"
+          comment="why do you want to learn 📚 ?"
+          errors={errors}
+          value={Object.keys(roles)[0]}
         >
-          {name}
-        </NavLink>
-      </span>
-    ))}
-  </Div>
-)
+          {Object.entries(roles).map(([key, { name, color }]) => (
+            <option value={key} style={{ background: color, color: 'white' }}>
+              {name}
+            </option>
+          ))}
+        </Select>
+      </Form>
+
+      {/* <P class="title-learning">What do you want to learn 📚 ?</P>
+      {Object.entries(roles).map(([key, { name, color }]) => (
+        <span class="navlink-learning" key={key}>
+          <NavLink
+            href={`${API}/link/discord?speciality=${key}`}
+            style={{ color: parseColor(color), fontSize: 35 }}
+          >
+            {name}
+          </NavLink>
+        </span>
+      ))} */}
```

Comment:
don't commit commented code. if you're find the better way to do what do you want in the simple way, you can remove the older code.